crates/broker/src/order_picker.rs

```diff
@@ -464,10 +510,37 @@ where
     /// Estimate of gas for locking a single order
     /// Currently just uses the config estimate but this may change in the future
     async fn estimate_gas_to_lock(&self, _order: &Order) -> Result<u64> {
+        // TODO: Add gas costs for orders with smart contract signatures.
```

Comment:
Hmm, we should probably set a gas limit when we call isValidSignature in the contracts, and use that here

Reply:
I've completed these changes now